### PR TITLE
Support dynamic schema (#24062) (#24176)

### PR DESCRIPTION
**Pull request by xiaocai2333**
/kind improvement
1. support dynamic schema for collection.
2. `$meta` is the default field name of dynamic field, can not use it as field name or json key.
3. Indexing of json field is not yet supported.
4. json key must enclosed in double quotes， unless you take the key to directly access.
    a. `JSONField["key"] > 10` or `key > 10` both are supported.

### CODE REVIEW

**Comment by xiaocai2333** [comment]
rerun ut

File: internal/rootcoord/create_collection_task.go [hunk 1]
```diff
@@ -101,8 +101,13 @@ func (t *createCollectionTask) validateSchema(schema *schemapb.CollectionSchema)
 	if t.Req.GetCollectionName() != schema.GetName() {
 		return fmt.Errorf("collection name = %s, schema.Name=%s", t.Req.GetCollectionName(), schema.Name)
 	}
-	if hasSystemFields(schema, []string{RowIDFieldName, TimeStampFieldName}) {
-		return fmt.Errorf("schema contains system field: %s, %s", RowIDFieldName, TimeStampFieldName)
+	if hasSystemFields(schema, []string{RowIDFieldName, TimeStampFieldName, MetaFieldName}) {
+		log.Error("schema contains system field",
```
Comment (czs007) [inline internal/rootcoord/create_collection_task.go:105]: better to change to Info, you can modify this in next pr

Comment (xiaocai2333) [reply internal/rootcoord/create_collection_task.go]: fixed

**Pull request merged**